### PR TITLE
Use generic simulator

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR* [version update](https://semver.org/)

### Context

No longer requires a specific Simulator device to be available, as we target generic devices (for example: `generic/platform=iOS Simulator`).
Applied new Input config standards.

### Changes

- _New input_: `destination` input, to specify a generic device: 
Destination specifier describes the device to use as a destination. Can be one of: `generic/platform=iOS Simulator`, `generic/platform=watchOS Simulator`, ` generic/platform=tvOS Simulator`.

  - `destination` replaces the now removed `simulator_device`, `simulator_os_version` and `simulator_platform`.

- _New input_: Build settings (`xcconfig_content`). Allows you to override the project's build settings. Creates a temporary file with the given input value as content and passes it to xcodebuild as the -xcconfig parameter.
  - Removed `code_signing_allowed` in favor of `xcconfig_content`. To enable code signing, set CODE_SIGNING_ALLOWED=YES in the `xcconfig_content` Input.
  - Removed `disable_index_while_building`.
- Renamed `is_clean_build` to `perform_clean_action`.
- Renamed `output_tool` to `log_formatter`. Can be `xcpretty` or `xcodebuild` as before.

### CODE REVIEW

File: step.go [hunk 1]
```diff
@@ -7,29 +7,32 @@ import (
...
-	"github.com/bitrise-io/xcode-project/xcscheme"
-	"github.com/bitrise-io/xcode-project/xcworkspace"
-	"github.com/bitrise-steplib/steps-xcode-archive/utils"
 	"github.com/bitrise-steplib/steps-xcode-build-for-simulator/util"
 	"github.com/kballard/go-shellquote"
 )
 
 const (
-	minSupportedXcodeMajorVersion = 7
+	minSupportedXcodeMajorVersion = 11
```
Comment (ofalvai) [inline step.go:35]: Do we still need this? Maybe the feature we check is available in Xcode 11 so we can completely remove the check.

Comment (lpusok) [reply step.go]: Removed.

**Comment by lo1tuma** [comment]
This is exactly what I’m looking for. Is there anything I can do to help getting this feature released?

**Review by godrei — APPROVED** [review]

**Comment by lpusok** [comment]
@lo1tuma @nasirky Released as version 1.0.0.

**Pull request merged**